lib/chef/provider/osx_profile.rb

```diff
@@ -0,0 +1,254 @@
+        @current_resource = Chef::Resource::OsxProfile.new(@new_resource.name)
+        @current_resource.profile_name(@new_resource.profile_name)
+
+        all_profiles = get_installed_profiles
+        @new_resource.profile(
+          @new_resource.profile ||
+          @new_resource.profile_name
+        )
+
+        @new_profile_hash = get_profile_hash(@new_resource.profile)
+        @new_profile_hash['PayloadUUID'] =
+          config_uuid(@new_profile_hash) if @new_profile_hash
+
+        if @new_profile_hash
+          @new_profile_identifier = @new_profile_hash['PayloadIdentifier']
+        else
+          @new_profile_identifier = @new_resource.identifier ||
+          @new_resource.profile_name
+        end
+
+        current_profile = all_profiles['_computerlevel'].find {
+          |item| item['ProfileIdentifier'] ==
+             @new_profile_identifier
+        }
+        @current_resource.profile(current_profile)
+
+      end
+
+      def define_resource_requirements
+        requirements.assert(:remove) do |a|
+          if @new_profile_identifier
+            a.assertion {
+              !@new_profile_identifier.nil? and
+              !@new_profile_identifier.end_with?('.mobileconfig') and
+              /^\w+(?:\.\w+)+$/.match(@new_profile_identifier)
+            }
+            a.failure_message RuntimeError, "when removing using the identifier attribute, it must match the profile identifier"
+          else
+          new_profile_name = @new_resource.profile_name
+            a.assertion {
+              !new_profile_name.end_with?('.mobileconfig') and
+              /^\w+(?:\.\w+)+$/.match(new_profile_name)
+            }
+            a.failure_message RuntimeError, "When removing by resource name, it must match the profile identifier "
+          end
+        end
+
+        requirements.assert(:install) do |a|
+          if @new_profile_hash.is_a?(Hash)
+            a.assertion {
+              @new_profile_hash.include?('PayloadIdentifier')
+            }
+            a.failure_message RuntimeError, "The specified profile does not seem to be valid"
+          end
+          if @new_profile_hash.is_a?(String)
+            a.assertion {
+              @new_profile_hash.end_with?('.mobileconfig')
+            }
+            a.failure_message RuntimeError, "#{new_profile_hash}' is not a valid profile"
+          end
+        end
+      end
+
+      def action_install
+        unless profile_installed?
+          converge_by("install profile #{@new_profile_identifier}") do
+            profile_path = write_profile_to_disk
+            install_profile(profile_path)
+            get_installed_profiles(update=true)
+          end
+        end
+      end
+
+      def action_remove
+        # Clean up profile after removing it
+        if profile_installed?
+          converge_by("remove profile #{@new_profile_identifier}") do
+            remove_profile
+            get_installed_profiles(update=true)
```

Comment:
This is a bug. You're passing in the result of `update = true` - ruby doesn't have the same kwargs that python does. This technically works only because the result of an assignment is that assignment:

```
irb(main):016:0> puts f='one'
one
```

But kwargs don't work:

```
irb(main):012:0> def foo(bar=nil, baz=nil)
irb(main):013:1> puts baz
irb(main):014:1> end
=> :foo
irb(main):015:0> foo(baz='one')
```